src/structures/Message.js

```diff
@@ -390,7 +390,7 @@ class Message extends Base {
    * @readonly
    */
   get guild() {
-    return this.client.guilds.resolve(this.guildId);
+    return this.client.guilds.resolve(this.guildId) ?? this.channel?.guild ?? null;
```

Comment:
Is this change necessary with the constructor's assignment?

Reply:
Likelihood? No
Theres an extremely small niche use where you fetch a message, the channel is not cached, now you have no guiildId, channel gets cached, now you have guild